src/main/scala/nodes/learning/KernelRidgeRegression.scala

```diff
@@ -0,0 +1,343 @@
+package nodes.learning
+
+import workflow.LabelEstimator
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+import edu.berkeley.cs.amplab.mlmatrix.RowPartitionedMatrix
+import edu.berkeley.cs.amplab.mlmatrix.util.{Utils => MLMatrixUtils}
+
+import scala.util.Random
+
+class KernelRidgeRegression(
+  kernelBlockGenerator: KernelGenerator,
```

Comment:
okay after thinking about this a bit I think the kernelBlockGenerator should still be passed into krr.scala (maybe not kernelModel). There are certain optimizations I do/computations I cache by calling KernelBlockGenerator inside the loop. Also it makes the cacheKernel argument kind of confusing since the user could have materialized the kernel outside the call into KRR. Maybe we can talk about this in person?